### DISCUSSION TITLE
Question about repmat() and inv_pos() on sdpvar

### DISCUSSION THREAD

**Discussion by dollylly** [yalmip/YALMIP#1142] [Problems using YALMIP] [answered]
I have question that x is a sdpvar variable with shape of 5*1. But when I write log2(1+ repmat(pj,4,1) .* inv_pos( repmat(pj,4,1))) and  expect a 5*4 matrix, but the result turns to be a 4*1 matrix as the following.  How can I fix it please?  Thanks!

K>> pj = sdpvar(1,4)
Linear matrix variable 1x4 (full, real, 4 variables)
Coeffiecient range: 1 to 1
K>> log2(1+ repmat(pj,4,1) .* inv_pos( repmat(pj,4,1)))
Nonlinear matrix variable 4x1 (full, real, 4 variables)
Coeffiecient range: 1.4427 to 1.4427
K>> repmat(pj,4,1)
Linear matrix variable 4x4 (full, real, 4 variables)
Coeffiecient range: 1 to 1

**Comment by johanlofberg** [comment, marked as answer]
Didn't I answer this elsewhere, I recognize the question?

Anyway, inv_pos is not a YALMIP function (it is a cvx operator). Hence, you will have to come up with a YALMIP specific way to model the thing you want to model. Unclear what you intend to do though, as the expression inside the log2 operator is constant as you basically do 1 + pj./pj

**Reply by johanlofberg** [reply]
https://github.com/yalmip/YALMIP/discussions/1143
